src/simulator_core/adapter/transforms/esdl_asset_mapper.py

```diff
@@ -17,13 +19,20 @@ class EsdlAssetMapper(EsdlMapperAbstract):
     }
 
     def to_esdl(self, entity: AssetAbstract) -> Any:
-        """ Maps entity object to PyEsdl objects"""
+        """Maps entity object to PyEsdl objects."""
         raise NotImplementedError("EsdlAssetMapper.to_esdl()")
 
-    def to_entity(self, model: EsdlAssetObject) -> AssetAbstract:
+    def to_entity(self, model: EsdlAssetObject, **kwargs) -> AssetAbstract:
         """Method to map an esdl asset to an asset entity class.
+
         :param EsdlAssetObject model: Object to be converted to an asset entity.
-        :return: Entity object. """
+        :return: Entity object.
+        """
         if not type(model.esdl_asset) in self.conversion_dict:
             raise NotImplementedError(str(model.esdl_asset) + ' not implemented in conversion')
-        return self.conversion_dict[type(model.esdl_asset)]()
+        return self.conversion_dict[type(model.esdl_asset)](model.esdl_asset.name,
+                                                            model.esdl_asset.id,
+                                                            kwargs['from_junction'],
+                                                            kwargs['to_junction'],
+                                                            kwargs['thermal_production'],
+                                                            kwargs['temperature_supply'])
```

Comment:
If you're retrieving the required data from the kwargs param, there's no guarantee that these parameters are actually available (resulting in an error/exception from the wrong location).    These should be fixed parameters in the EsdlAssetObject

Reply:
Refactor to only pass name and id